### ISSUE TITLE
[Bug] DisplaySnackBarAsync Task never completes if another SnackBar appears

### ISSUE THREAD

**Issue by Dreamescaper** [xamarin/XamarinCommunityToolkit#1341]
### Steps to Reproduce

Execute the following:
```csharp
            var task1 = page.DisplaySnackBarAsync("Test1", "Action", () => Task.Delay(10));
            var task2 = page.DisplaySnackBarAsync("test2", "Action", () => Task.Delay(10));

            var res1 = await task1;
            var res2 = await task2;
```

### Expected Behavior
Both tasks are completed, res1 and res2 are false;

### Actual Behavior
task1 never completes, therefore execution is stuck.

### Basic Information

- Version with issue: 1.1.0
- Platform Target Frameworks:
  - Android

**Comment by pictos** [comment]
@Dreamescaper 
Could you attach a small repro?

**Comment by Dreamescaper** [comment]
Sure!
I've attached a sample project. Start the project, click on "Show multiple SnackBars" button.

Expected:
When all snackbars are hidden (due to timeout), button handler completes, and prints 'Completed';

Actual:
Execution is stuck on `await snackbarTask1` line, it never completes.

[Repro.zip](https://github.com/xamarin/XamarinCommunityToolkit/files/6566269/Repro.zip)

**Comment by maxkoshevoi** [comment]
@Dreamescaper Thanks for the sample project! It helped a lot with the investigation.
Issue will be fixed after #1357 is merged.

![image](https://user-images.githubusercontent.com/6609929/120538374-19a86b00-c3ef-11eb-8d52-cccef39e470b.png)

You can test the solution yourself. NuGet is it pipeline artifacts.
